CMakeLists.txt

```diff
@@ -1,19 +1,51 @@
-project(StatsdClient)
-cmake_minimum_required(VERSION 3.2)
-
+# Basic project setup
+cmake_minimum_required(VERSION 3.5)
+project(cpp-statsd-client
+        VERSION 1.0.2
+        LANGUAGES CXX
+        DESCRIPTION "A header-only StatsD client implemented in C++"
+        HOMEPAGE_URL "https://github.com/vthiery/cpp-statsd-client")
 set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_EXTENSIONS OFF)
-set(CPP_STATSD_CLIENT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include/cpp-statsd-client)
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+
 
-include_directories(${CPP_STATSD_CLIENT_DIR})
+# The library target
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(
+  ${PROJECT_NAME}
+  INTERFACE $<BUILD_INTERFACE:${${PROJECT_NAME}_SOURCE_DIR}/include>
+            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
-install(DIRECTORY ${CPP_STATSD_CLIENT_DIR} DESTINATION include)
+# The installation and pkg-config-like cmake config
+install(TARGETS ${PROJECT_NAME}
+        EXPORT ${PROJECT_NAME}_Targets
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+write_basic_package_version_file("${PROJECT_NAME}ConfigVersion.cmake"
+                                 VERSION ${PROJECT_VERSION}
+                                 COMPATIBILITY SameMajorVersion)
+configure_package_config_file(
+  "${PROJECT_SOURCE_DIR}/cmake/${PROJECT_NAME}Config.cmake.in"
+  "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+  INSTALL_DESTINATION
+  ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/cmake)
+install(EXPORT ${PROJECT_NAME}_Targets
+        FILE ${PROJECT_NAME}Targets.cmake
+        NAMESPACE ${PROJECT_NAME}::
+        DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/cmake)
+install(FILES "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
+              "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+        DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/${PROJECT_NAME}/cmake)
+install(DIRECTORY ${PROJECT_SOURCE_DIR}/include DESTINATION include)
 
+# The test targets
 find_package(Threads)
 add_executable(testStatsdClient ${CMAKE_CURRENT_SOURCE_DIR}/tests/testStatsdClient.cpp)
-target_link_libraries(testStatsdClient ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(testStatsdClient ${CMAKE_THREAD_LIBS_INIT} ${PROJECT_NAME})
```

Comment:
having the test target depend on the library target makes it so that modifying the latter will trigger recompilation of the the former. previously you could get yourself in a state where changes wouldnt make the test target need to be rebuilt